storage_scrubber/src/lib.rs

```diff
@@ -230,13 +230,58 @@ pub fn remote_timeline_path_id(id: &TenantShardTimelineId) -> RemotePath {
 
 #[derive(Debug, Clone, Serialize, Deserialize)]
 #[serde(deny_unknown_fields)]
-pub struct BucketConfig {
+pub enum BucketConfig {
+    Legacy(BucketConfigLegacy),
```

Comment:
I'd prefer to eliminate this Legacy variant: could we just always have a regular RemoteStorageConfig, but for the S3-only code paths, extract the bucket+region from it and assert out if it's not an S3 config?

Reply:
done